src/mono/System.Private.CoreLib/src/System/RuntimeType.Mono.cs

```diff
@@ -1615,6 +1615,34 @@ private void CreateInstanceCheckThis()
             return CreateInstanceMono(false, true);
         }
 
+        // Specialized version of the above for Activator.CreateInstance<T>()
+        [DebuggerStepThroughAttribute]
+        [Diagnostics.DebuggerHidden]
+        internal void CallDefaultStructConstructor(ref byte value)
+        {
+            Debug.Assert(IsValueType);
+
+            RuntimeConstructorInfo? ctor = GetDefaultConstructor();
+            if (ctor == null)
+                return;
```

Comment:
`CreateInstanceT_StructWithPrivateDefaultConstructor_ThrowsMissingMethodException` is failing on Mono, perhaps due to this being a `return` instead of a `throw`

Reply:
It is a bit more complicated than that, but thanks for point out the issue. If the `ctor` is `null`, that means it isn't defined, which turns into a nop. However, if the `ctor` is defined, it must be `public`. That second check is what I am missing.